File: iosApp/iosApp/Presentation/Screens/Articles/Coordinators/ArticlesCoordinator.swift
//
//  ArticlesCoordinator.swift
//  iosApp
//
//  Created by a.kottaridis on 29/7/24.
//  Copyright © 2024 orgName. All rights reserved.
//

import UIKit
import shared

public class ArticlesCoordinator: Coordinator {
    
    public override func start() {
        super.start()
        goToArticlesLanding()
    }
}

extension ArticlesCoordinator: ArticlesActionHandler {
    
    public func goToArticlesLanding() {
        let viewModel = ArticlesViewModel(
            useCase: ArticlesInjector().articlesUseCase,
            actionHandler: self
        )
        let wrapper = ArticlesViewModelWrapper(viewModel: viewModel)
        let screen = ArticlesScreen(viewModel: wrapper)
        navigate(
            to: screen,
            with: .push,
            hideNavigationBar: true,
            hideTabBar: false,
            isSwipeBackEnable: true
        )
    }
    
    public func goToAboutScreen() {
        let screen = AboutScreen()
        navigate(
            to: screen,
            with: .push,
            hideNavigationBar: true,
            hideTabBar: false,
            isSwipeBackEnable: true
        )
    }
    
    public func goToArticleDetails(articleId: String) {
        // TODO: - Add Screen Implementation
    }
}

File: androidApp/src/main/java/com/thanasis/kottaridis/thannewskmm/android/presentation/screen/articles/ArticlesCoordinator.kt
package com.thanasis.kottaridis.thannewskmm.android.presentation.screen.articles

import android.widget.Toast
import com.thanasis.kottaridis.thannewskmm.android.application.MainActivity
import com.thanasis.kottaridis.thannewskmm.android.application.base.navigation.Coordinator
import com.thanasis.kottaridis.thannewskmm.presentation.articles.ArticlesActionHandler

class ArticlesCoordinator :
    Coordinator<MainActivity>(),
    ArticlesActionHandler {
    override fun start() {
        goToArticlesLanding()
    }

    /**
     * ArticlesActionHandler Implementation
     */
    override fun goToArticlesLanding() {
        Toast.makeText(safeActivity, "Articles Coordinator goToArticlesLanding", Toast.LENGTH_SHORT).show()
    }

    override fun goToArticleDetails(articleId: String) {
        Toast.makeText(safeActivity, "Articles Coordinator goToArticleDetails", Toast.LENGTH_SHORT).show()
    }

    override fun goToAboutScreen() {
        Toast.makeText(safeActivity, "Articles Coordinator goToAboutScreen", Toast.LENGTH_SHORT).show()
    }
}
